### PR TITLE
VTOL Front transition minimum time

### PR DESCRIPTION
**Pull request by sanderux**
Fixes https://github.com/PX4/Firmware/issues/3850
Minimum transition time for front transition for standard vtol

### CODE REVIEW

File: src/modules/vtol_att_control/standard.cpp [hunk 1]
```diff
@@ -153,6 +158,7 @@ void Standard::update_vtol_state()
 			// start transition to fw mode
 			_vtol_schedule.flight_mode = TRANSITION_TO_FW;
 			_vtol_schedule.transition_start = hrt_absolute_time();
+			warnx("Front transition started");
```
Comment (LorenzMeier) [inline src/modules/vtol_att_control/standard.cpp]: Debug output

File: src/modules/vtol_att_control/standard.cpp [hunk 2]
```diff
@@ -164,12 +170,17 @@ void Standard::update_vtol_state()
 
 		} else if (_vtol_schedule.flight_mode == TRANSITION_TO_FW) {
 			// continue the transition to fw mode while monitoring airspeed for a final switch to fw mode
-			if (_airspeed->indicated_airspeed_m_s >= _params_standard.airspeed_trans || !_armed->armed) {
+			if ((_airspeed->indicated_airspeed_m_s >= _params_standard.airspeed_trans &&
+			     (float)hrt_elapsed_time(&_vtol_schedule.transition_start)
+			     > (_params_standard.front_trans_time_min * 1000000.0f)) ||
+			    !_armed->armed) {
+				warnx("Front transition complete");
```
Comment (LorenzMeier) [inline src/modules/vtol_att_control/standard.cpp]: Debug output

File: src/modules/vtol_att_control/standard.cpp [hunk 3]
```diff
@@ -164,12 +170,17 @@ void Standard::update_vtol_state()
...
 				_vtol_schedule.flight_mode = FW_MODE;
 				// we can turn off the multirotor motors now
 				_flag_enable_mc_motors = false;
 				// don't set pusher throttle here as it's being ramped up elsewhere
 				_trans_finished_ts = hrt_absolute_time();
+				warnx("Front transition complete");
```
Comment (LorenzMeier) [inline src/modules/vtol_att_control/standard.cpp]: Debug output

**Comment by LorenzMeier** [comment]
Other than the remaining debug output looking good!

**Comment by sanderux** [comment]
I think this output is very useful and would like to leave it in, it now only says "Forwarding command 3000" and doesnt say anything about back transition. Any chance we could leave this in?

**Comment by sanderux** [comment]
I do see it outputs the front trans compete twice. If you agree to let the other 2 in i will remove the redundant one.

**Comment by LorenzMeier** [comment]
Please mark them with XXX or TODO. The console is not accessible in flight anyway.

**Comment by sanderux** [comment]
Logging solved in https://github.com/PX4/Firmware/pull/3869

**Comment by LorenzMeier** [comment]
Is this flight tested in SITL or outdoors?

**Comment by sanderux** [comment]
Sitl

**Comment by LorenzMeier** [comment]
@AndreasAntener If this is flight tested we could bring it in I guess.

**Comment by AndreasAntener** [comment]
Yes, but it doesn't look like we can test this week. @sanderux ?

**Comment by sanderux** [comment]
I might, bit windy

**Comment by sanderux** [comment]
Flight tested on an actual failing airspeed sensor:
http://logs.uaventure.com/view/2prHQ9Mp8PGV3qcJJ2LWSM

Here it shows this param actually saved the transition.

**Comment by AndreasAntener** [comment]
Nice, thanks for testing! Rebased, waiting on checks.

**Pull request merged**